BinaryEncoding.md

```diff
@@ -786,9 +783,140 @@ for [future :unicorn:][future multiple tables] use and must be 0 in the MVP.
 | `f32.reinterpret/i32` | `0xbe` | | |
 | `f64.reinterpret/i64` | `0xbf` | | |
 
+## Thread operators ([described here](Semantics.md#thread-operators))
+
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `is_lock_free` | `0xff00` | | |
+| `i32.wait` | `0xff01` | `memory immediate` | |
+| `i64.wait` | `0xff02` | `memory immediate` | |
+| `i32.wake` | `0xff03` | `memory immediate` | |
+| `i64.wake` | `0xff04` | `memory immediate` | |
+
+## Atomic read-modify-write operators ([described here](Semantics.md#atomic-memory-accesses))
+
+| Name | Opcode | Immediates | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.atomic.rmw.add` | `0xff10` | `0` `memory immediate` | atomic add to memory |
```

Comment:
> Why use memory immediate to denote the operation (add / sub / etc)? Why not use another opcode instead?

I considered using separate opcodes, but it eats up the opcode space pretty quickly since we have 12 variants per operation. I also kind of like that all the rmw ops are grouped together, but I suppose that's just personal preference.

> It would be helpful to do a map of opcode extension space usage here, similar to what was done for the MVP opcodes.

Sorry, do you mean mapping the usage of the atomics prefix space? Or mapping the usage of the remaining opcode space?

> I still think we should use 0xf0, it's free!

I don't have a preference for "atomics/threads" prefix, `0xf0` is just as good as anything else as far as I'm concerned. But it does seem like adding opcode prefixes starting at the end and going backward is logical. I suppose we could argue for 0xf0 by saying all prefixes start with `0xf_` and this is the first one.